scripts/build_obelisk.sh

```diff
@@ -1,28 +1,54 @@
 #!/bin/bash
 
-if [ -z "$PIXI_ENVIRONMENT_NAME" ]; then
-    if [ -z "$OBELISK_ROOT" ]; then
-        echo -e "/033[1;31mOBELISK_ROOT is not set. Run dev_setup.sh first!\033[0m"
-        exit 0
-    fi
-
-    echo -e "\033[1;32mBuilding Obelisk messages outside of a pixi env...\033[0m"
-    curr_dir=$(pwd)
-    cd $OBELISK_ROOT/obelisk_ws
-    colcon build --symlink-install --parallel-workers $(nproc) \
-        --packages-select obelisk_control_msgs obelisk_estimator_msgs obelisk_sensor_msgs obelisk_std_msgs
-    source $OBELISK_ROOT/obelisk_ws/install/setup.bash
-    colcon build --symlink-install --parallel-workers $(nproc) \
-        --packages-skip obelisk_control_msgs obelisk_estimator_msgs obelisk_sensor_msgs obelisk_std_msgs
-    source $OBELISK_ROOT/obelisk_ws/install/setup.bash
-    cd $curr_dir
-
-else
-    echo -e "\033[1;32mBuilding Obelisk messages within a pixi env...\033[0m"
-    pixi run messages-build
-    source $OBELISK_ROOT/obelisk_ws/install/setup.bash
-    pixi run ros-build
-    source $OBELISK_ROOT/obelisk_ws/install/setup.bash
+leap=false
+zed=false
+
+for arg in "$@"; do
+    case $arg in
+        # hardware options
+        --leap)
+            leap=true
+            shift  # Adds leap ROS packages to colcon build command
+            ;;
+        --zed)
+            zed=true
+            shift  # Adds ZED ROS packages to colcon build command
+            ;;
+        *)
+            # Unknown option
+            echo "Unknown option: $arg"
+            echo "Usage: $0 [--leap] [--zed]"
+            exit 1
+            ;;
+    esac
+done
+
+MESSAGE_PKGS=" obelisk_control_msgs obelisk_estimator_msgs obelisk_sensor_msgs obelisk_std_msgs"
+
+# configuring which packages to skip
+SKIP_PKGS=""
+if [ "$leap" = false ]; then
+    SKIP_PKGS+=" obelisk_leap_cpp obelisk_leap_py"
+fi
+if [ "$zed" = false ]; then
+    SKIP_PKGS+=""  # TODO(ahl): fill this once ZED packages are added
```

Comment:
Pretty sure you've added the ZED packages in so you should be able to fill this out. Still reading the code so I'll see for sure later - just dropping this here so I don't forget.

Reply:
i haven't yet, those are referring to the ZED ROS2 packages (follow-up PR). I should clarify this comment, though.